src/PKSim.Core/Services/IndividualSimulationEngine.cs

```diff
@@ -64,6 +69,13 @@ public async Task RunAsync(IndividualSimulation individualSimulation, Simulation
          }
       }
 
+      private void addEvents(ISimModelManager simModelManager)
+      {
+         if (simModelManager == null) return;
+         simModelManager.SimulationProgress += simulationProgress;
+         simModelManager.Terminated += terminated;
+      }
```

Comment:
Not sure about removing the events after the simModelManager is done, but it couldn't hurt.